Added 'Orientation' parameter for vertical and horizontal layouts

fhaefele opened the pull request:
![test_horizontal](https://github.com/bastibe/Violinplot-Matlab/assets/74400839/73a8032e-6aeb-4d39-a815-409e57b18b75)
![example](https://github.com/bastibe/Violinplot-Matlab/assets/74400839/c60ce474-b9a5-4636-9b0e-5aa9245507d2)
![example2](https://github.com/bastibe/Violinplot-Matlab/assets/74400839/2e8bf7fe-d30f-45c9-a292-2480a1a5fb06)

bastibe commented on Violin.m:
```diff
@@ -1,714 +1,726 @@
+                        jitter = -1*rand(size(data2));% left
+                        [x,y] = obj.plotOrientation(pos + jitter.*jitterstrength,data2);
+                        obj.ScatterPlot2 = ...
+                            scatter(x, y, args.MarkerSize, 'filled');         
+                    else
+                        [x,y] = obj.plotOrientation(pos + jitter.*jitterstrength,data);
+                        obj.ScatterPlot = ...
+                            scatter(x, y, args.MarkerSize, 'filled');
+
+                    end
+                case 'histogram'
+                    [counts,edges] = histcounts(data, size(unique(data),1));
+                    switch args.HalfViolin
+                        case 'right'
+                            [x,y] = obj.plotOrientation([pos-((counts')/max(counts))*max(jitterstrength)*2, pos*ones(size(counts,2),1)]',...
+                                        [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2]);    
+                        case 'left'
+                            [x,y] = obj.plotOrientation([pos*ones(size(counts,2),1), pos+((counts')/max(counts))*max(jitterstrength)*2]', ...
+                                        [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2]);
+                        otherwise
+                            fprintf([namefun, ' No histogram/bar plot option available for full violins, as it would look overcrowded.\n'])
+                    end
+                    obj.HistogramPlot = plot(x,y,'-','LineWidth',1, 'Color', 'k');
```
This is referring to an earlier comment.

Here, the `case`s no longer plot, but merely prepare an `x` and `y`. This makes it harder to understand what's going on. I'd prefer the plot commands to still be inlined, as this is explicitly what the code does: It plots things. Preparing data is merely an ancillary complexity for the dual-orientation use case.

This is exacerbated by the name `plotOrientation`, which implies that the function actually plots. Which it doesn't. It merely swaps two variables.

I'd prefer a structure like this in every `case` instead:

```
x = [pos*ones(size(counts,2),1), pos+((counts')/max(counts))*max(jitterstrength)*2]';
y = [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2];
[x, y] = obj.swapOrientationsMaybe(x, y);
obj.HistogramPlot = plot(x, y, '-', 'LineWidth', 1, 'Color', 'k');
```

Our first objective here is readability and maintainability!

bastibe replied:
I would use the above function name `swapOrientationsMaybe`. But feel free to use a different name if you find one.

bastibe reviewed (changes requested):
Thank you for the contribution! This is super useful!

However, I don't like the way the `plotOrientation` function makes the code harder to reason about in some places. I have sketched out an alternative structure that I'd prefer.

Nevertheless, great work, and thank you very much!

fhaefele commented on Violin.m:
```diff
@@ -1,714 +1,726 @@
+        ViolinPlotQ     % fill plot of the Quartiles as shadow
+    end
+
+    properties (Hidden = true)
```
Sounds good. I will fix it.

fhaefele committed:
Added orientation parameter which allows to plots to generate a vertical or horizontal layout.

fhaefele committed:
Orientation property is set to read only. Adjustments to code readability and maintainability

fhaefele committed:
Merge branch 'master' into violin_orientation

fhaefele committed:
Orientation property is set to read only. Adjustments to code readability and maintainability

fhaefele commented:
Hm, merging and resolving conflicts is still pretty new to me. I gave it my best. Also I have addressed all of your comments. Let me know if everything fits now, else I will try to fix it asap. I was also thinking of including that 'parent' property which you discussed in the other PR.

bastibe commented on Violin.m:
```diff
@@ -1,714 +1,733 @@
+            results = p.Results;
+        end
+
+        function [x,y] = swapOrientationMaybe(obj, x, y)
```
could you add a quick documentation to the function?

bastibe reviewed (changes requested):
This looks much better, thank you!

Just one more quick note: Could you format the code with spaces after operators (such as commas), so it matches the rest of the code?

Otherwise, this looks good to me!

fhaefele committed:
Added documentation to swapOrientationMaybe function and code formatting.

fhaefele commented:
Should be done and ready to merge. Do you want to also add the features from #24 ? Should be an easy inclusion.

bastibe commented:
> Do you want to also add the 'parent' property from https://github.com/bastibe/Violinplot-Matlab/pull/24 ?

That would be a good idea, thank you!

fhaefele committed:
Added 'Parent' property to plot violins on specific axes.

fhaefele commented:
I have added now the parent property and checked the functionality of the orientation swap and the 'parent' axis in a modified `testviolinplot.m`. To me it looks like it works fine.

bastibe commented:
Looks good to me! Thank you! 

If you want to, you could add an example to the README.

Is this ready to merge on your end?

bastibe commented:
I'll leave it up to you whether you want to add it to the README.

fhaefele committed:
adjusted README and readme_figures. Small code snippet works now on its own.

fhaefele committed:
test cases 9 and 10 added.

fhaefele commented:
I have edited the README and the `readme_figures.m` respectively. The README now shows a code snippet that can work on its own, which the previous one couldn't. I have also added two new test cases which test the 'Parent' and 'Orientation' property. Feel free to merge it into the main branch as you like.

bastibe commented:
Beautiful! Thank you!
